api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/MetricHandler.java

```diff
@@ -74,26 +77,31 @@
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "Successfully retrieved at least one metric definition."),
             @ApiResponse(code = 204, message = "No metrics found."),
-            @ApiResponse(code = 400, message = "Missing or invalid type parameter type.", response = ApiError.class),
+            @ApiResponse(code = 400, message = "Invalid type parameter type.", response = ApiError.class),
             @ApiResponse(code = 500, message = "Failed to retrieve metrics due to unexpected error.",
                          response = ApiError.class)
     })
     public void findMetrics(
             @Suspended final AsyncResponse asyncResponse,
             @ApiParam(value = "Queried metric type",
-                      required = true,
+                      required = false,
                       allowableValues = "[gauge, availability, counter]")
-            @QueryParam("type") MetricType metricType
-    ) {
-        if (metricType == null) {
-            asyncResponse.resume(badRequest(new ApiError("Missing type param")));
+            @QueryParam("type") MetricType metricType,
+            @ApiParam(value = "List of tags", required = false) @QueryParam ("tags") Tags tags) {
+        if(metricType != null && !MetricType.userTypes().stream().filter(t -> metricType == t)
```

Comment:
How about `userTypes().contains(metricType)` ? This stream+filter+findAny construct looks a bit convoluted in this case.